File: python/easy/split_a_string_in_balanced_strings.py
# 1221. Split a String in Balanced Strings
# https://leetcode.com/problems/split-a-string-in-balanced-strings/


class Solution:
    def balancedStringSplit(self, s: str) -> int:
        r_count, l_count, substring = 0, 0, 0
        for i in s:
            if i == "R":
                r_count = r_count + 1
            elif i == "L":
                l_count = l_count + 1
            
            if r_count > 0 and l_count > 0 and r_count == l_count:
                substring = substring + 1
                r_count, l_count = 0, 0
        
        return substring

File: javascript/easy/split_a_string_in_balanced_strings.js
// 1221. Split a String in Balanced Strings
// https://leetcode.com/problems/split-a-string-in-balanced-strings/

var balancedStringSplit = function(s) {
    var counts = {
        "r_count": 0,
        "l_count": 0,
        "substring": 0
    };
    
    for (var i = 0; i < s.length; i++) {
        if ( s[i] == "R" ) {
            counts["r_count"]++;
        } else if ( s[i] == "L" ) {
            counts["l_count"]++;
        }
        
        if ( (counts["r_count"] > 0) && (counts["l_count"] > 0) && (counts["r_count"] == counts["l_count"]) ) {
            counts["substring"]++;
        }
    }
    
    return counts["substring"];
};